### DISCUSSION TITLE
Can you use ytdl-org/youtube-dl#options to download videos partially

### DISCUSSION THREAD

**Discussion by blockbabak** [community/community#37276] [Other Feature Feedback, Questions, & Ideas]
Hi,
Is there a way to download a short sample of a video file by which you can determine the codec information about it without downloading the whole video?
If not using ytdl-org then some other tool.

**Comment by airtower-luna** [comment]
Try the `-F` option, it should list the available formats with their codecs.

**Reply by blockbabak** [reply]
That gives you the option of downloading formats.
What I want to know is the codec used for the file without downloading it. Is it H.264 or something else?
That kind of information is usually in the file header. But if you could download part of the file as a complete file then you can get the information.
 Youtube used to embed such info about the file in the source code of the page but that is gone.

**Reply by airtower-luna** [reply]
> What I want to know is the codec used for the file without downloading it. Is it H.264 or something else?

That depends on the format you choose to download, which is exactly the point. Pick the format you want to download (or if you use the default, check what `youtube-dl --get-format` reports), and combine it with the list that `-F` provides.

**Comment by blockbabak** [comment]
The point is that I want to avoid downloading the file. I have a long list (1000) of videoid(s) and I want to know the information about them without downloading them at this point because that would take a very, very long time
This information used to be embedded in the youtube page but it is no longer there. So, I am trying to find a way round it.

**Comment by airtower-luna** [comment]
None of what I suggested would download the file. My point is that `-F` lists the *available* formats *with their codecs*, so you can can simply look up the codec(s) for the format you're interested in. Mind that there are often (depending on the video site) *multiple* formats with different codecs, so there won't be the one and only codec for a video (unless you're looking at a video hoster that offers only one).

**Comment by otter-computer** [comment]
Hi @blockbabak,

Thanks for being a part of the GitHub Community! We're glad you're here.

If you're looking for help for this specific topic, you might want to try asking for help somewhere that focuses on this project, such as the [project's repository](https://github.com/ytdl-org/youtube-dl/issues). It's possible that another GitHub user might have run into this same issue and can help, but the GitHub Community on Discussions focuses primarily on topics related to GitHub itself or collaboration on project development and ideas. We want to make sure you’re getting the best support you can, but this space may not be the right place for this topic.

Best of luck!
